chroot_setup.py

```diff
@@ -0,0 +1,373 @@
+def chroot_system(line: str):
+  line = line.rstrip()
+  print('system -i "' + line + '"')
+  if (CHROOT_DEBUG == False):
+    os.system('system -i "' + line + '"')
+
+# TODO: documentation
+def chroot_sh(line: str):
+  print(line)
+  if (CHROOT_DEBUG == False):
+    os.system(line)
+
+action_dictionary = {
+  ':file'       : chroot_setup,
+  ':mkidr'      : chroot_mkdir,      # done
+  ':ln_fix_rel' : chroot_ln_fix_rel, # done
+  ':ln_rel'     : chroot_ln_rel,     # done
+  ':ln'         : chroot_ln,         # done
+  ':mknod'      : chroot_mknod,      # done
+  ':cp_dir'     : chroot_cp_dir,     # done
+  ':cp'         : chroot_cp,         # done
+  ':chmod_dir'  : chroot_chmod_dir,  # done
+  ':chmod'      : chroot_chmod,      # done
+  ':chown_dir'  : chroot_chown_dir,  # done
+  ':chown'      : chroot_chown,      # done
+  ':system'     : chroot_system,
+  ':sh'         : chroot_sh
+}
+
+def chroot_setup (chroot_lst):
+  lst_file = open(chroot_lst, 'r')
+  action = None
+
+  # Iterate over all lines in the file, executing actions where encountered
+  for line in lst_file:
+    line = line.strip()
+
+    # Blank line
+    if not line:
+      # print('empty')
+
+    # Comment
+    else if line.startswith('#'):
+      # print('comment')
+
+    # Determine action
+    else if line.startswith(':'):
+      action = action_dictionary[line]
+
+    # Execute action
+    else:
+      if action not None:
+        action(line)
+
+# TODO: documentation
+def initial_check ():
+  if (os.path.exists('/QOpenSys/usr/bin')):
+    CHROOT_DEBUG = False
+    PATH='/QOpenSys/usr/bin:/QOpenSys/usr/sbin:'
+    LIBPATH='/QOpenSys/usr/lib'
+    os.environ['PATH'] = PATH
+    os.environ['LIBPATH'] = LIBPATH
+    print("**********************")
+    print("Live IBM i session (changes made).")
+    print("**********************")
+
+    if (ags.verbose):
+      print('PATH=' + PATH)
+      print('LIBPATH=' + LIBPATH)
+
+  else:
+    CHROOT_DEBUG = True
+    print("**********************")
+    print("Not IBM i, no action is taken (debug flow purpose only).")
+    print("**********************")
+    # exit -1
+
+
+def yum_install(package_list: list):
+  os.environ["LIBPATH"] = '/QOpenSys/pkgs/lib:/QOpenSys/usr/lib'
+  install_template = Template('/QOpenSys/pkgs/bin/yum $yes --installroot=$chroot_dir install $package')
+  auto_yes = '-y' if args.auto_yes else ''
+  for package in package_list:
+    command = install_template.substitute(yes=auto_yes, chroot_dir=args.chroot_directory, package=package)
+    logger.debug(command)
+    os.system(command)
+
+#
+# MAIN
+#
+
+lst = 0
+ops = 0
+qsys = 0
+CHROOT_DIR = ""
+CHROOT_LIST = ""
+SCRIPT_DIR = os.path.dirname(os.readlink(args.chroot_directory)) + "/config"
+
+# Didn't implement all of the checks, argparse can do that
+
+intial_check()
+
+# This is done because the arg may be a relative path name.
+# Need to be done before any 'cd'
+TEMP_DIR = os.readlink(args.chroot_directory)
+
+logger.debug("Scipt Dir: " + SCRIPT_DIR)
```

Comment:
The [default log level](https://docs.python.org/3.6/howto/logging.html#basic-logging-tutorial) is set to `WARNING` therefore all the logger.debug calls won't print unless the log level is adjusted. We could check if `-v` verbose mode was set and adjust the log level to something else i.e. `INFO` or `DEBUG`

```python
import logging
logging.warning('Watch out!')  # will print a message to the console
logging.info('I told you so')  # will not print anythinglogging.warning('Watch out!')  # will print a message to the console
logging.info('I told you so again!')  # will not print anything
```